tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -388,19 +389,34 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
         OptimizerHints.empty, targetMDef.hash)
   }
 
-  private def findInheritedMethodDef(classInfo: Analysis.ClassInfo,
-      methodName: String, getTree: TreeProvider,
+  private def findInheritedMethodDef(analysis: Analysis,
+      classInfo: Analysis.ClassInfo, methodName: String, getTree: TreeProvider,
       p: Analysis.MethodInfo => Boolean = _ => true): MethodDef = {
     @tailrec
     def loop(ancestorInfo: Analysis.ClassInfo): MethodDef = {
       assert(ancestorInfo != null,
           s"Could not find $methodName anywhere in ${classInfo.encodedName}")
 
       val inherited = ancestorInfo.methodInfos.get(methodName)
-      if (inherited.exists(p)) {
-        findMethodDef(ancestorInfo, methodName, getTree)
-      } else {
-        loop(ancestorInfo.superClass)
+      inherited.find(p) match {
+        case Some(m) =>
+          m.syntheticKind match {
+            case MethodSyntheticKind.None =>
+              findMethodDef(ancestorInfo, methodName, getTree)
+
+            case MethodSyntheticKind.DefaultBridge(targetInterface) =>
+              val targetInterfaceInfo = analysis.classInfos(targetInterface)
+              findMethodDef(targetInterfaceInfo, methodName, getTree)
```

Comment:
Can you make an example when we actually have to follow a default bridge? IIUC, the analyzer would resolve the reflective call to the default method directly to the interface. So the target of such a reflective proxy would be the default method implementation, not a bridge...